File: VB/WindowsApplication3/Main.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors
Imports WindowsApplication3.Data

Namespace WindowsApplication3
    Partial Public Class Main
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
            recordBindingSource.DataSource = DataHelper.GetData(100)
            imageComboBoxEdit.Properties.Items.AddEnum(GetType(FilterPanelLocation))
            imageComboBoxEdit.EditValue = FilterPanelLocation.Top

            treeList1.ActiveFilterString = "[ID] > 1"
        End Sub

        Private Sub OnFilterPanelLocationChanged(ByVal sender As Object, ByVal e As EventArgs) Handles imageComboBoxEdit.SelectedIndexChanged
            treeList1.FilterPanelLocation = CType(imageComboBoxEdit.EditValue, FilterPanelLocation)
        End Sub
    End Class
End Namespace

File: CS/WindowsApplication3/Main.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using WindowsApplication3.Data;

namespace WindowsApplication3 {
    public partial class Main : XtraForm {
        public Main() {
            InitializeComponent();
            recordBindingSource.DataSource = DataHelper.GetData(100);
            imageComboBoxEdit.Properties.Items.AddEnum(typeof(FilterPanelLocation));
            imageComboBoxEdit.EditValue = FilterPanelLocation.Top;

            treeList1.ActiveFilterString = "[ID] > 1";
        }

        private void OnFilterPanelLocationChanged(object sender, EventArgs e) {
            treeList1.FilterPanelLocation = (FilterPanelLocation)imageComboBoxEdit.EditValue;
        }
    }
}
